docs/shadow_paths.md

```diff
@@ -0,0 +1,81 @@
+    OperStatus       E_Interface_OperStatus             `path:"state/oper-status" module:"openconfig-interfaces/openconfig-interfaces"`
+}
+```
+
+The aim of this document is to clarify what shadow paths are and why they exist
+as a generation option.
+
+## Compressed-out "config" or "state" YANG leaves
+
+Shadow paths are relevant only for compressed GoStructs, and indicate the
+compressed-out `/config` or `/state` YANG `leaf` nodes.
+
+https://datatracker.ietf.org/doc/html/draft-openconfig-netmod-opstate-01#section-2
+contains a diagram of the relationship between intended config (`/config`) and
+applied config (`/state`) leaves:
+
+```
+          +---------+
+          |         |    transition intended
+          |intended |    to applied
+          | config  +---------+
+          |         |         |
+          +---------+         |
+              ^               |         config: true
+   +----------|------------------------------------+
+              |               |         config: false
+              |               |
+              |               |
+              |       +-----------------------------+
+              |       |       | operational state   |
+              |       |  +----v----+ +-----------+  |
+              |       |  |         | |           |  |
+              +       |  | applied | |  derived  |  |   operational:true
+            same +------>| config  | |   state   |<-------+
+            leaves    |  |         | |           |  |
+                      |  |         | |           |  |
+                      |  +---------+ +-----------+  |
+                      +-----------------------------+
+```
+
+In this diagram, applied config are "config false", or `/state` leaves that
+mirror intended config leaves, which are "config true", or `/config` leaves. Per
+[design.md](design.md#openconfig-path-compression), one of these are compressed
+out depending on the value of the `-prefer_operational_state` generation flag.
+
+## Problem with Path Compression
+
+Path compression, however, leads to the GoStruct not being able to represent
+both intended config and applied config at the same time, but for certain use
+cases (e.g. [ygnmi](https://github.com/openconfig/ygnmi#queries)), it is
+desirable to use the same GoStruct for representing either the (intended
+config+derived state) combination, as well as the (applied config+derived state)
+combination of leaves. This is where shadow paths can help.
+
+## Preferring Shadow Paths
+
+This term is used to describe utilities preferring the `shadow-path` tag instead
+of the `path` tag in the generated GoStructs when they both exist on a field,
+and is therefore used to switch the meaning of the GoStruct from a "config view"
+(intended config+derived state) to a "state view" (applied config+derived
+state).
+
+For example, if a gNMI update for `/interfaces/interface[name="foo"]/state/mtu`
```

Comment:
maybe give the converse example where a config update is ignored when ygot.PreferShadowPath is false?

Reply:
I didn't do this exactly but expanded on this example and added a description on this "ignoring" behaviour in the previous section.